From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 22 Oct 2021 01:57:29 +0200
Subject: [PATCH] Build with PulseAudio backend only

PipeWire (the preferred audio server nowadays) provides PulseAudio API,
there's IMHO not much reason for building Spot with both PulseAudio and
ALSA backends.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,8 @@
 
 [dependencies.librespot]
 version = "0.3.1"
-features = ["alsa-backend", "pulseaudio-backend"]
+default-features = false
+features = ["pulseaudio-backend"]
 
 [dependencies.protobuf]
 version = "2.8.1"
--- a/data/dev.alextren.Spot.gschema.xml
+++ b/data/dev.alextren.Spot.gschema.xml
@@ -2,7 +2,6 @@
 <schemalist gettext-domain="spot">
   <enum id="dev.alextren.Spot.AudioBackend">
     <value value="0" nick="pulseaudio"/>
-    <value value="1" nick="alsa"/>
   </enum>
   <enum id="dev.alextren.Spot.Bitrate">
     <value value="0" nick="96"/>
@@ -37,10 +36,6 @@
     <key name='audio-backend' enum='dev.alextren.Spot.AudioBackend'>
       <default>'pulseaudio'</default>
       <summary>Audio backend</summary>
-    </key>
-    <key name='alsa-device' type='s'>
-      <default>'default'</default>
-      <summary>Alsa device (if audio backend is 'alsa')</summary>
     </key>
   </schema>
 </schemalist>
